demos/args/README.md

```diff
@@ -0,0 +1,45 @@
+# Args
+
+In this demo we are going to show that we can create threads with a worker routine and pass at most 4 arguments to them.
+
+To describe the demo, we will introduce the following:
+
+* `k_add_thread` (in `src/kernel/thread.c`)
+* `thread_start` (in `src/kernel/thread.c`)
+
+## `k_add_thread`
+
+The functions like `add_named_thread`, `add_thread_from_worker`, and `add_named_thread_with_args` are mapped to syscall `add_thread`. So we will jump to this function when we call them.
+
+This function checks if the `args` we gave is `NULL`. If it is `NULL` the function creates a dummy `ThreadArgs` with 4 zeroed arguments then uses it as the new thread's arguments.
+
+## `thread_start`
+
+`thread_start` is the starting point of every thread. When the scheduler chooses a thread to run, it loads the arguments from the thread structure and runs the worker routine.
+
+## Walkthrough
+
+For starters, `setup` thread creates named threads `printer` and `subprinter` with several arguments.
+
+The values we pass to `printer` and the corresponding arguments are listed below:
+
+* `3` -> `repeat`
+* `"aardvark"` -> `phrase`
+* `2` -> `sub_printer`
+* `2` -> `start`
```

Comment:
This would make more sense if we reverse the order. Like:

We pass the following arguments to `printer`:
* `repeat` = `3`